### DISCUSSION TITLE
Conductor UI build - Yarn to NPM : Issues resolved

### DISCUSSION THREAD

**Discussion by shekarpcs** [Netflix/conductor#2772] [Announcements]
I have been working on building Conductor ui using NPM instead of Yarn. Initially It looked pretty straight forward but I had vis-data issues. 

Failed to compile.

./node_modules/react-vis-timeline/node_modules/vis-timeline/esnext/esm/vis-timeline-graph2d.js
Attempted import error: 'isDataViewLike' is not exported from 'vis-data/esnext/esm/vis-data.js'.

After searching in internet with multiple trial and errors following solutions worked. 

in package.json:

Add following in dependencies: 
"vis-data": "7.1.0" 

Next Modify following :

 "resolutions": {
    "vis-timeline": "^7.4.2",
    "vis-timeline": "7.3.6"
  },

Run following:

for local:
> npm install
> npm start

For production build:

npm run build

We copied generated build folder to to nginx server. Everything is working as expected. 

Posting this for reference for others.

**Comment by peterlau** [comment]
Thanks for the suggestion. We'll investigate deeper and see whether this change can be incorporated into the official build.
